SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1946,13 +1965,13 @@ - (nullable NSDictionary *)throws_encryptedMessageForMessageSend:(OWSMessageSend
     
     // Loki: Both for friend request messages and device link messages we use fallback encryption as we don't necessarily have a session yet
     BOOL isFriendRequest = [messageSend.message isKindOfClass:LKFriendRequestMessage.class];
-    BOOL isDeviceLinkMessage = [messageSend.message isKindOfClass:LKDeviceLinkMessage.class];
-    if (isFriendRequest || (isDeviceLinkMessage && ((LKDeviceLinkMessage *)messageSend.message).kind == LKDeviceLinkMessageKindRequest)) {
-        return [self throws_encryptedFriendRequestOrDeviceLinkMessageForMessageSend:messageSend deviceId:@(OWSDevicePrimaryDeviceId) plainText:plainText];
-    }
+    BOOL isDeviceLinkMessage = [messageSend.message isKindOfClass:LKDeviceLinkMessage.class] && ((LKDeviceLinkMessage *)messageSend.message).kind == LKDeviceLinkMessageKindRequest;
+//    if ((isDeviceLinkMessage && ((LKDeviceLinkMessage *)messageSend.message).kind == LKDeviceLinkMessageKindRequest)) {
```

Comment:
@RyanRory can you comment on this change?

Reply:
A device link message is basically a friend request, with the sealed sender we don't need to encrypt the message here, we move the encryption process for a friend request message inside the UD sender message encryption process. These codes can be deleted actually. If we keep these, the envelope will be built with a type of friend request and the source cannot be hidden.

Reply:
That makes sense, cheers :)